graylog2-web-interface/src/logic/rest/FetchProvider.js

```diff
@@ -152,29 +149,29 @@ function queuePromiseIfNotLoggedin(promise) {
   return promise;
 }
 
-export default function fetch(method, url, body) {
+export default function fetch(method, url, body, handleForbidden, handleUnauthorized) {
   const promise = () => new Builder(method, url)
     .authenticated()
-    .json(body)
+    .json(body, handleForbidden, handleUnauthorized)
     .build();
 
   return queuePromiseIfNotLoggedin(promise)();
 }
 
-export function fetchPlainText(method, url, body) {
+export function fetchPlainText(method, url, body, handleForbidden, handleUnauthorized) {
   const promise = () => new Builder(method, url)
     .authenticated()
-    .plaintext(body)
+    .plaintext(body, handleForbidden, handleUnauthorized)
     .build();
 
   return queuePromiseIfNotLoggedin(promise)();
 }
 
-export function fetchPeriodically(method, url, body) {
+export function fetchPeriodically(method, url, body, handleForbidden, handleUnauthorized) {
   const promise = () => new Builder(method, url)
     .authenticated()
     .noSessionExtension()
-    .json(body)
+    .json(body, handleForbidden, handleUnauthorized)
     .build();
 
```

Comment:
@dennisoelkers A few questions regarding these changes:
Would you add e.g. `logoutIfUnauthorized ` as a Builder method? And would you allow a customisation of the whole part or only the condition check? (in the context of `logoutIfUnauthorized` and `redirectIfForbidden `)